wolfcrypt/src/aes.c

```diff
@@ -13360,25 +13599,34 @@ int wc_AesEaxEncryptFinal(AesEax* eax, byte* authTag, word32 authTagSz)
     word32 cmacSize;
     int ret;
     word32 i;
+    int ciphertextCmac_finalized = 0;
+    int aadCmac_finalized = 0;
 
-    if (eax == NULL || authTag == NULL || authTagSz > AES_BLOCK_SIZE) {
+    if (eax == NULL) {
         return BAD_FUNC_ARG;
     }
 
+    if (authTag == NULL || authTagSz > AES_BLOCK_SIZE) {
+        ret = BAD_FUNC_ARG;
+        goto out;
```

Comment:
Not sure about this change in behavior to free on an error with bad input rather than allowing the user to get the return value, see that a bad argument was found, and attempt calling again with corrected arguments.

Reply:
Yeah similar story to `wc_CmacFinal()`, though as you point out, this change has the potential to break existing (but broken) client code.

My goal doing it this way was to make a solid attempt at safely eliminating leaks for existing client code, i.e. without the client code having to add explicit error handling code and a call to a new `wc_AesEaxFree()` API.  My assumption is that no actual existing client code is _actually_ retrying the call with a different `authTag` and/or `authTagSz` than the first call, which I think is a realistic assumption, given that this is the finalizing call, not an initializing call that might realistically be subject to iterative trial to discover a supported keysize.